lib/app.js

```diff
@@ -294,13 +294,20 @@ function uploadSite(settings, directoryRef, argv) {
 }
 
 function getSettings(argv) {
-  var settingsFile = path.resolve('./firebase.json');
-  var settingsJSON, settings;
-  if (!fs.existsSync(settingsFile)) {
-    console.log(chalk.red('Initialization Error') + ' - Directory not ' +
+  var curDir = process.cwd();
+  var settingsFile;
+  while(!fs.existsSync((settingsFile = (curDir + '/firebase.json')))) {
```

Comment:
also, in addition, we'd put a space after control keywords like `while`, `if`, `catch` etc